### PR TITLE
Pipe MountPoint to audit logs

### PR DESCRIPTION
**Pull request by clemon**
[VAULT-15882](https://hashicorp.atlassian.net/browse/VAULT-15882)
required for Vault Insights, this PR simply pipes the existing `Request.MountPoint` into audit logs

[VAULT-15882]: https://hashicorp.atlassian.net/browse/VAULT-15882?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by clemon** [comment]
- vault server in dev mode, audit logs enabled
- `./vault kv put -mount=secret chettest foo=bar`
- audit log contains `mount_point`
```
{
  "time": "2023-04-27T21:13:03.726637Z",
  "type": "response",
  "auth": {
    [...]
  },
  "request": {
    "mount_point": "secret/",
    "mount_type": "kv",
    "mount_accessor": "kv_4f00a322",
    "mount_running_version": "v0.14.2+builtin",
    "mount_class": "secret",
    [...]
  },
  "response": {
    "mount_point": "secret/",
    "mount_type": "kv",
    "mount_accessor": "kv_4f00a322",
    "mount_running_plugin_version": "v0.14.2+builtin",
    "mount_class": "secret",
    [...]
    }
  }
}
```

**Comment by clemon** [comment]
via user-defined path
- `./vault secrets enable -path=chetkv kv`
- `./vault kv put -mount=chetkv chettest foo=bar`
- audit log contains `mount_point` chetkv
```
{
  "time": "2023-04-27T21:17:33.390072Z",
  "type": "response",
  "auth": {
    [...]
  },
  "request": {
    "mount_point": "chetkv/",
    "mount_type": "kv",
    "mount_accessor": "kv_f0cfb6c8",
    "mount_running_version": "v0.14.2+builtin",
    "mount_class": "secret",
    [...]
  },
  "response": {
    "mount_point": "chetkv/",
    "mount_type": "kv",
    "mount_accessor": "kv_f0cfb6c8",
    "mount_running_plugin_version": "v0.14.2+builtin",
    "mount_class": "secret"
  }
}
```

**Review by ncabatoff — APPROVED** [review]

**Comment by clemon** [comment]
via user-generated subpath
```
{
  "time": "2023-04-27T21:22:10.659441Z",
  "type": "response",
  "auth": {
    [...]
  },
  "request": {
    "mount_point": "chetkv2/subpath/",
    "mount_type": "kv",
    "mount_accessor": "kv_a9420101",
    "mount_running_version": "v0.14.2+builtin",
    "mount_class": "secret",
    [...]
  },
  "response": {
    "mount_point": "chetkv2/subpath/",
    "mount_type": "kv",
    "mount_accessor": "kv_a9420101",
    "mount_running_plugin_version": "v0.14.2+builtin",
    "mount_class": "secret"
  }
}
```

File: changelog/20411.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:improvement
+audit: pipes the existing "Request.MountPoint" into audit logs
```
Comment (ncabatoff) [inline changelog/20411.txt:2]:
```suggestion
audit: add a `mount_point` field to audit requests and response entries
```

**Review by swayne275 — APPROVED** [review]
Eyyyyy first Vault PR! Great work!

**Comment by swayne275** [comment]
would you mind backporting this to version 1.12.x also? i know I didn't specify that at the beginning, but it looks like clusters that currently have insights are running v1.12.x

**Pull request merged**